hudi-common/src/main/java/org/apache/hudi/common/data/HoodieData.java

```diff
@@ -220,6 +220,15 @@ <O> HoodieData<O> mapPartitions(SerializableFunction<Iterator<T>,
    */
   HoodieData<T> repartition(int parallelism);
 
+  /**
+   * Coalesces underlying collection (if applicable) making sure new {@link HoodieData} has
+   * exactly {@code parallelism} partitions or less.
+   *
+   * @param parallelism target number of partitions in the underlying collection
+   * @return {@link HoodieData<T>} holding coalesced collection
+   */
+  HoodieData<T> coalesce(int parallelism);
```

Comment:
This looks like it is only used in a spark flow right now. Do we need to add it to the general HoodieData interface?